### ISSUE TITLE
Feature: Inventory bots

### ISSUE THREAD

**Issue by Tolfx** [TF2Autobot/tf2autobot#1483]
**Is your feature request related to a problem? Please describe.**
Currently bots can have a max cap of 3k slots. If it exceeds well nothing much you can do than hope somethings get bought or withdraw from it.

**Describe the feature you'd like**
I suggest there could also be a inventory settings to make some bots act as inventory for a main bot so when it exceeds it can put perhaps items there and make the inventory bot, perhaps do task with the stuff it holds.

Not sure how it would really work, more of a concept I had that would be nice to have but implementing might not work as well.

**Comment by zudsniper** [comment]
> ...
> @Tolfx 

While this may sound simple, in reality this would require implementing essentially an entire barebones botnet. While I like the idea, I don't think it would fit within this project -- if anything, it would likely fork it or sit atop it. Interesting angle of attack though, specifying inventory bots vs. trading bots. Feel free to reach out if you need help. C:

**Comment by Tolfx** [comment]
> > ...
> > @Tolfx
> 
> While this may sound simple, in reality this would require implementing essentially an entire barebones botnet. While I like the idea, I don't think it would fit within this project -- if anything, it would likely fork it or sit atop it. Interesting angle of attack though, specifying inventory bots vs. trading bots. Feel free to reach out if you need help. C:

I totally agree is not suited for this project, is perhaps more of an extension that is built from the project instead. I think it wouldn't be that hard, but a lot of refactoring of the code needs to be done to make it possible. Especially how the setup is done right now.

Also need a structure layout on communication on each bot, since we want to have a shared knowledge of each bot and make the best decision based around that.

So the simplest way of doing it would probably make the storage bots just wait for trade offers from master bot, which it will accept, which can be either a withdraw or deposit. Our master should also retrieve their current inventory so we can make correct trade offers depending on what we need to do.

A scenario could look like on a really really simple solution:
1. Master bot requires 100 ref.
2. Sends a broadcast message across all bots for their inventory and receives.
3. Storage bot 1 has 300 ref and bot 2 has 20 ref.
4. We want our trade to go through as fast as possible
  4.1 We send a trade offer from bot 1 of 80 ref
  4.2 We send a trade offer from bot 2 of 20 ref
5. Storage bots accepts the trade.

This approach would probably be really easy, ofc not only about refined, could also be about keys, would have a pulse that always ask their inventory which we store in memory so if would need something we can request for it, almost including it as our own inventory.

Is pretty narrow approach tho, since we don't know if want to have more master bots that shares storage bots. So a lot of thought can go into it.

Think the approach to this right now would to make a proof of concept and see if can make some sort of logic like this.
